WeatherMonitoringSystem/Domain/Utilities/UserInteraction.cs

```
@@ -0,0 +1,103 @@
+﻿
+using WeatherMonitoringSystem.Bots;
+using WeatherMonitoringSystem.Core;
+using WeatherMonitoringSystem.DataParsers;
+
+namespace WeatherMonitoringSystem.UI;
+
+public static class UserInteraction
+{
+
+    public static void Run()
+    {
+        PrintMessage("Welcome to the Real-Time Weather Monitoring System!");
+
+        WeatherParserStrategy parserStrategy = new WeatherParserStrategy();
+
+        while (true)
+        {
+            string format = GetDataFormat();
+
+            try
+            {
+                IWeatherDataParser parser = GetWeatherDataParser (format);
+
+                parserStrategy.SetParser(parser);
+               
+            }
+            catch (Exception ex)
```

`Exception` is too generic, it can be internal error messages that we shouldn't show to the user. We would usually catch specific exceptions (i.e. argument exception) that we know we will be thrown from the logic called, and we can leave the last cash to be for `Exception` and not show `ex.Message` to the user but instead something more generic `Something went wrong`. As generic exceptions could hold sensitive information.

If we don't know what type of exceptions the code would throw, we can catch `Exception` but instead of printing `ex.Message` we can do a custom exception `Something went wrong` and log the exception somewhere (we don't have to log anything now since we are not at that place in the training yet :D )

Lets revisit other try-catch blocks in this PR with similar behavior and fix :)